File: Day2/167-Two-Sum-II-/Method2.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int n = nums.length;

        for (int i = 0; i < n; i++) {
            int val1 = nums[i];
            int complement = target - val1;

            int low = 0, high = n - 1;
            while (low <= high) {
                int mid = low + (high - low) / 2;
                if (complement == nums[mid] && mid != i)
                    return new int[] { i + 1, mid + 1 };
                else if (complement < nums[mid])
                    high = mid - 1;
                else
                    low = mid + 1;
            }
        }

        return new int[] {};
    }
}

File: Day2/724-Find-Pivot-Index/Method2.cpp
class Solution {
public:
    int pivotIndex(vector<int>& nums) {
        int n = nums.size();
        int sum[n];
        int leftSum = 0;

        for (int i = n - 1; i >= 0; i--) {
            if (i == n - 1) {
                sum[i] = nums[i];
                continue;
            }
            sum[i] = sum[i + 1] + nums[i];
        }

        for (int i = 0; i < n; i++) {
            leftSum += nums[i];
            if (leftSum == sum[i])
                return i;
        }

        return -1;
    }
};